resources/lib/vrtplayer/vrtapihelper.py

```diff
@@ -160,19 +160,27 @@ def _map_to_episode_items(self, episodes, titletype=None):
             ))
         return episode_items, sort
 
-    def _map_to_season_items(self, api_url, buckets, episode):
+    def _map_to_season_items(self, api_url, seasons, episode):
         season_items = []
         fanart = statichelper.add_https_method(episode.get('programImageUrl'))
+        program_type = episode.get('programType')
         metadata_creator = metadatacreator.MetadataCreator()
         metadata_creator.mediatype = 'season'
-        for bucket in buckets:
-            season_title = bucket.get('key')
-            title = ''.join((self._kodi_wrapper.get_localized_string(32094), ' ', season_title))
-            season_facet = '&facets[seasonTitle]='
-            if ' ' in season_title:
-                season_title = season_title.replace(' ', '-')
+        # Reverse sort seasons if program_type is 'reeksaflopend' or 'daily'
+        seasons = sorted(seasons, key=lambda k: k['key'], reverse=True) if program_type in ('reeksaflopend', 'daily') else seasons
+        for season in seasons:
+            season_key = season.get('key')
+            season_episode_count = str(season.get('doc_count'))
+            title = ''.join((self._kodi_wrapper.get_localized_string(32094), ' ', season_key))
+            # API workaround: season_key matches seasonTitle facet, but seasonTitle behaves as a starting match regex
+            # and therefore sometimes mixes up episodes from different seasons, instead we prefer to use seasonName facet
+            # which behaves as an exact match regex and in most cases matches season_key when spaces are replaced by dashes.
+            if len(episode.get('seasonTitle')) > 1 and episode.get('seasonTitle').replace(' ', '-') == episode.get('seasonName'):
```

Comment:
Not sure if the `len > 1` is a good identifier. Can we not do i.e. `int()` instead ?

Reply:
Not sure what you mean. `seasonTitle` is a string.

Reply:
I get that, but I assumed that you do `len(episode.get('seasonTitle')) > 1` to filter out integer values as seasonTitle. So I wonder if turning it into an integer wouldn't be a better check.

Reply:
No, not necessary, I found out that just checking the season data of one episode is not failsafe when VRT uses a different `seasonName` naming convention for the other seasons.
This was the case with "Down the road" https://vrtnu-api.vrt.be/search?i=video&size=150&facets[programName]=down-the-road
So, I tried to filter out as much values that are not affected by the "starting match regex problem" and luckily this fixes the "Down the road" problem.

Reply:
I don't understand what this has to do with a length of more than 1 though. So I have a big problem if we create conditional statements without a clue to what it relates to. It makes it impossible to refactor in the future not knowing if you break something that worked before.

I assumed you wanted to filter out when seasonTitle was 1-9, now I don't know why you do this. A season could be 10 or 11 too.

Reply:
From the subtitle you can see the VRT is using the seasonTitle and not the seasonName, so I guess we need to do the same. Somehow I think the solution is in the displayOptions.

PS I discovered it's not actually a "starting match regex", for `2018-2019` the following match: `2018`, `201`, `2`, but also `2019`. What doesn't work is `19` or `18`. A lot of special characters seems to terminate the string (including `-`, `+`, `*` or `|`) although if any special character is repeated before the [A-z0-9] string, these characters are ignored. A very weird implementation...

Reply:
Values with a length of one are immune for the "starting match regex problem". I filter them out so they are handled by the `seasonTitle` facet.

I don't have a better solution right now to fix "Down the road" season listing:  https://vrtnu-api.vrt.be/search?i=video&size=150&facets[programName]=down-the-road
Down the road has two seasons with `seasonTitle` values: `1` and `2` and `seasonName` values: `2018` and `2`
From `season.get('key')` we get  `1` and `2`  as input values.

So, the most logical thing is to use `seasonTitle` facet here.

Reply:
Well, that is why I suggested to check if they are integers instead, or maybe even check if the value is `[A-z0-9]+`.

Reply:
I don't understand, for instance value `1` and value `13` are affected by the "starting match regex problem"

Reply:
Ok, so my concern here is that if we *have to* implement workarounds for very specific corner-cases, we need to clearly identify these corner-cases in comments (with actual examples) or we have no chance in the future to refactor/improve this code ever. (Ideally we would have proper unit-tests for this at some point).

Now that I understand a bit better why this is more complex, let me suggest some changes in this PR.

PS I'd rather avoid these complexities, but it's clear we have no clue how to escape special characters (which seem to terminate the string matching) at this time, if this is at all possible :-/

Reply:
Feel free to fix this better. I don't have time this week to look at this again.

Reply:
The "better" fix will mostly be documenting the corner-cases, but I need to test them myself first, and that won't happen this week...

Reply:
It seems seasonTitle facet is implemented in the Elasticsearch backend as a partial match. This is not fixable in a good way using api requests. I fixed this again in a different way.